### PR TITLE
A3.5 HTTP Post Call Handling

### PR DESCRIPTION
**Pull request by tdang2180**
#26 
- Implemented POST call handling and added tests for name/nickname/pronouns.

### CODE REVIEW

File: mesh/tests/profiles_tests.py [hunk 1]
```diff
@@ -89,6 +89,21 @@ def test_no_account_preferred_pronouns(self):
...
+        user_name_data = {"userName": "kwame brown"}
+        response = self.client.post(f'/profiles/user-name/{self.test_profile.accountID}', user_name_data)
+        self.assertEquals(response.status_code, 200)
+
+    def test_post_preferred_name (self):
+        preferred_name_data = {"preferredName": "brown"}
+        response = self.client.post(f'/profiles/preferred-name/{self.test_profile.accountID}', preferred_name_data)
+        self.assertEquals(response.status_code, 200)
+
+    def test_post__preferred_pronouns (self):
```
Comment (shotoh) [inline mesh/tests/profiles_tests.py:102]: Pretty sure you didn't mean to put two underscores here

**Review by shotoh — APPROVED** [review]
Everything else looks fine to me

File: mesh/tests/profiles_tests.py (hunk 1 above)
Comment (tdang2180) [reply mesh/tests/profiles_tests.py:102]: 👍

File: mesh/profiles/views.py [hunk 2]
```diff
@@ -77,3 +85,31 @@ def get_data(account_id, name, mapper):
             "status": "error",
             "message": "An account does not exist with this account ID."
         }, status=404)
+
+def post_data(account_id, name, request):
+    """
+    Handles POST requests when the client sends names/nicknames/prounouns to the back end
+    """
+    try: 
+        account = Account.objects.get(accountID = account_id)
```
[the remark this answers is not in the document]
Comment (tdang2180) [reply mesh/profiles/views.py:94]: Good point, I removed it. 👍

File: mesh/profiles/views.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 from django.core.exceptions import ObjectDoesNotExist
+from django.utils.datastructures import MultiValueDictKeyError
 from django.http import JsonResponse
 
 from mesh.accounts.models import Account
 from mesh.profiles.models import Profile
-
+from mesh.accounts.models import Account
```
Comment (JRSiwiecki) [inline mesh/profiles/views.py:7]: Probably remove one of these duplicate imports.

File: mesh/profiles/views.py [hunk 4]
```diff
@@ -77,3 +85,27 @@ def get_data(account_id, name, mapper):
...
+
+        if (name == "userName"):
+            profile.userName = data 
+        elif (name == "preferredName"):
+            profile.preferredName = data
+        elif (name == "preferredPronouns"):
+            profile.preferredPronouns = data
+    
+        profile.save()
+        return JsonResponse({'message': f'{name} saved successfully'}, safe=False, status = 200)
```
Comment (JRSiwiecki) [inline mesh/profiles/views.py:105]: I think here, the `safe=False` parameter can be removed, as the `safe` parameter only needs to be set to false if we are sending objects that are not JSON serializable, but in this case, you are just returning a string in the message, so `safe = False` shouldn't be necessary (though I would still test it to see if I'm correct or not). 

Point being that I don't see why `safe = False` would be needed here but not on line 111, which is nearly identical to this one.

**Review by JRSiwiecki — CHANGES_REQUESTED** [review]
Everything looks good for the most part, I just added some small suggestions. Re-ping me on discord when you get a chance to either make changes or provide feedback!

File: mesh/profiles/views.py (hunk 4 above)
Comment (tdang2180) [reply mesh/profiles/views.py:105]: Yeah, you are right. I removed it 👍.

**Review by JRSiwiecki — APPROVED** [review]
Good work 👍🏼

**Pull request merged**